### ISSUE TITLE
powershell provisioner fails when username or password contains a $

### ISSUE THREAD

**Issue by akalman** [hashicorp/packer#5258]
Packer Version: 1.0.4
Host: OSX 10.12.6

```
Provisioning with shell script: /home/teamcity/tmp/buildTmp/packer-powershell-provisioner928700296
Opening /home/teamcity/tmp/buildTmp/packer-powershell-provisioner928700296 for reading
Building elevated command wrapper for: if (Test-Path variable:global:ProgressPreference){$ProgressPreference='SilentlyContinue'};$env:PACKER_BUILDER_TYPE="virtualbox-ovf"; $env:PACKER_BUILD_NAME="virtualbox-ovf"; &'c:/Windows/Temp/script.ps1';exit $LastExitCode
Uploading elevated shell wrapper for command [if (Test-Path variable:global:ProgressPreference){$ProgressPreference='SilentlyContinue'};$env:PACKER_BUILDER_TYPE="virtualbox-ovf"; $env:PACKER_BUILD_NAME="virtualbox-ovf"; &'c:/Windows/Temp/script.ps1';exit $LastExitCode] to [${env:TEMP}\packer-elevated-shell-5995f1f2-266d-3f37-9690-9caa6e6d1469.ps1] from [/home/teamcity/tmp/buildTmp/packer-elevated-shell.ps1244927303]
2975 bytes written for 'uploadData'
Uploading file to '${env:TEMP}\packer-elevated-shell-5995f1f2-266d-3f37-9690-9caa6e6d1469.ps1'
107 bytes written for 'uploadData'
Uploading file to 'c:/Windows/Temp/script.ps1'
starting remote command: powershell -executionpolicy bypass -file "%TEMP%\packer-elevated-shell-5995f1f2-266d-3f37-9690-9caa6e6d1469.ps1"
Exception calling "RegisterTaskDefinition" with "7" argument(s): "The user
name or password is incorrect. (Exception from HRESULT: 0x8007052E)"
At C:\Users\Administrator\AppData\Local\Temp\packer-elevated-shell-5995f1f2-266d-3f37-9690-9caa6e6d1469.ps1:49 char:1
$f.RegisterTaskDefinition($name, $t, 6, "Administrator", "somepasswordwitha$init", 1,  $null)
```

**Comment by mwhooker** [comment]
related to but I don't think exactly a duplicate of #3913

**Comment by SwampDragons** [comment]
Can I get a repro case (a gist with the packer.json and any other necessary files?) Are you using the ssh communicator or the winRM communicator?  When I try to repro with the winRM communicator, it won't even connect.

**Comment by akalman** [comment]
I no longer have access to the source code unfortunately but I was using the winRM communicator.  I can see if I can reproduce sometime in the next week or so but from what I recall the basic sequence was set the password using unattended, then spin up an elevated privileges powershell script using the powershell communicator.

**Comment by SwampDragons** [comment]
Thanks for the note; I'll try to figure out a repro case when I have a chance but if anyone else has more details for me that'd help me get to this faster.

**Comment by SwampDragons** [comment]
Ah, I wasn't escaping the $ correctly in the autounattend 🤦‍♀️

**Comment by SwampDragons** [comment]
Okay, repro'd and found a workaround. 
If anyone else comes across this, you need to escape the $ using a powershell escape character, which is a backtick.  So your password `Super$ecret` becomes ``Super`$ecret``

```
{
  "builders": [
  {
    "type": "amazon-ebs",
    "region": "us-east-1",
    "instance_type": "m3.medium",
    "source_ami": "ami-5ee10024",
    "ami_name": "packer-demo-{{timestamp}}",
    "user_data_file": "./boot_config/packer_bootstrap_win.txt",
    "communicator": "winrm",
    "winrm_username": "Administrator",
    "winrm_password": "Super$3cr3t!"
  }],
  "provisioners": [
    {
      "type": "powershell",
      "inline": "Write-Output(\"HELLO SWAMPY I AM POWERSHELL WOOOOOO\")",
      "elevated_user": "Administrator",
      "elevated_password": "Super`$3cr3t!"
    }
  ]
}
```

**Comment by SwampDragons** [comment]
This will get closed in 1.2.0 because the change will require some backwards-incompatible changes (will break passwords that have been properly powershell escaped).  In the meantime, I recommend user use the escaping workaround above.

**Comment by SwampDragons** [comment]
Closed by #5376

**Comment by dkorel-copperleaf** [comment]
i was on 1.2 and hitting this issue, upgraded to latest and it works.
